python/tank/bootstrap/baked_configuration.py

```diff
@@ -172,7 +172,7 @@ def bake_config_scaffold(cls, path, sg_connection, plugin_id, config_descriptor)
 
         config_writer.ensure_project_scaffold()
         config_descriptor.copy(os.path.join(path, "config"))
-        config_writer.install_core(config_descriptor, bundle_cache_fallback_paths=[])
+        config_writer.install_core(config_descriptor.resolve_core_descriptor())
```

Comment:
`install_core` used to be able to resolve the latest core if `core_api.yml` didn't specify a core version. Because the method now accepts the core descriptor directly, that had to be moved somewhere else. Considering baked configurations are static and should always hardcode the core version to use, it makes sense to assume that `resolve_core_descriptor` will never return `None`. Hence, this change is safe.

The code inside `install_core` to get the latest version was actually moved to CachedConfiguration, which it actually makes sense to have it.